crates/key-server/src/externals.rs

```diff
@@ -30,43 +26,28 @@ pub(crate) fn add_upgraded_package(pkg_id: ObjectID, new_pkg_id: ObjectID) {
 
 pub(crate) async fn fetch_first_pkg_id(
     pkg_id: &ObjectID,
-    network: &Network,
+    sui_client: &SuiClient,
 ) -> Result<ObjectID, InternalError> {
     match CACHE.get(pkg_id) {
         Some(first) => Ok(first),
         None => {
-            let query = serde_json::json!({
-                "query": format!(
-                    r#"
-                    query {{
-                        latestPackage(
-                            address: "{pkg_id}"
-                        ) {{
-                            address
-                            packageAtVersion(version: 1) {{
-                                address
-                            }}
-                        }}
-                    }}
-                    "#,
-                )
-            });
-
-            let response = Client::new()
-                .post(network.graphql_url())
-                .json(&query)
-                .send()
+            let object = sui_client
+                .read_api()
+                .get_object_with_options(*pkg_id, SuiObjectDataOptions::default().with_bcs())
                 .await
                 .map_err(|_| InternalError::Failure)?
-                .tap(|r| debug!("Graphql response: {:?}", r))
-                .json::<Value>()
-                .await
+                .into_object()
                 .map_err(|_| InternalError::Failure)?;
 
-            let first = response["data"]["latestPackage"]["packageAtVersion"]["address"]
-                .as_str()
-                .ok_or(InternalError::InvalidPackage)
-                .and_then(|s| ObjectID::from_str(s).map_err(|_| InternalError::Failure))?;
+            let package = object
+                .bcs
+                .ok_or(InternalError::Failure)?
+                .try_as_package()
+                .ok_or(InternalError::InvalidPackage)?
+                .to_move_package(u64::MAX)
+                .map_err(|_| InternalError::InvalidPackage)?;
```

Comment:
if we expect all packages to include to_move_package then this one should be Failure